src/worker/ingestion/process-event.ts

```diff
@@ -52,9 +49,8 @@ export class EventsProcessor {
         this.teamManager = new TeamManager(pluginsServer.db)
         this.personManager = new PersonManager(pluginsServer)
 
-        this.posthog = nodePostHog('sTMFPsFhdP1Ssg', { fetch })
         if (process.env.NODE_ENV === 'test') {
-            this.posthog.optOut()
+            posthog.optOut()
         }
```

Comment:
This now changes a globally exported var deep inside the event processing code. That feels a bit off... and if I look at shared/posthog, I'd never expect that I also need to look here to find some opt out logic.

Thus I think this check could easily go in shared/posthog.